CIPs/cip-137.md

```diff
@@ -18,190 +18,239 @@ An API for reading and writing events on Ceramic event streams.
 
 
 ## Motivation
 <!--Motivation is critical for CIPs that want to change the Ceramic protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the CIP solves. CIP submissions without sufficient motivation may be rejected outright.-->
-In order to enable databases other than ComposeDB to be built on top of Ceramic, an API to read and write directly on event streams is needed. This also enables developers to get direct access to the event streams underlaying ComposeDB.
+In order to enable services and databases other than ComposeDB to be built on top of Ceramic, an API to read and write directly on event streams is needed. This also enables developers to get direct access to the event streams underlaying ComposeDB.
 
 
 ## Specification
 <!--The technical specification should describe the syntax and semantics of any new feature.-->
 
-This is a work in progress specification that introduces a JSON-RPC interface for interacting directly with event streams on a node.
+This specification introduces a OpenAPI spec for interacting directly with event streams on a node.
 
 
 
-### `ceramic_subscribe`
+### GET `/ceramic/subscribe/{streamid}`
 
-This method tells the Ceramic node to subscribe to a given interest range of streams. A interest range is defined as `ceramic://*?<sort-key>=<sort-value>`, where *sort-key* describes a field in the header of the InitEvent of a stream, and *sort-value* is the value of this field. For example to subscribe to all MIDs of a particular Model you can use `ceramic://*?model=kjzl6hvfrbw6c82mkud4qs38zl4hd03ifoyg2ksvfjkhuxebfzh3ef89vwvtvrr`.
+This endpoint is called to subscribe to new events in a given interest range. This endpoint uses the node’s local ordering of events within the given range, constructed in the order in which the node received the events. If you want the global ordering of events you can use the *recon* endpoint. 
 
-Once subscribed the Ceramic node will use the Recon protocol to keep in sync with other nodes subscribed to this interest range.
+The range is normally defined as `ceramic://*?<sort-key>=<sort-value>`, where *sort-key* describes a field in the header of the InitEvent of a stream, and *sort-value* is the value of this field. For example to subscribe to all MIDs of a particular Model you can use `ceramic://*?model=kjzl6hvfrbw6c82mkud4qs38zl4hd03ifoyg2ksvfjkhuxebfzh3ef89vwvtvrr`.
 
-Active subscriptions need to be persisted. Or can be stateless somehow? (Nathaniel to share more thoughts here)
+This endpoint will respond as soon as the node knows about new events, given the *offset*, or when *duration* amount of seconds has passed. If the node already knows about a lot of new events, only *limit* amount of events will be returned.
```

Comment:
this is the first time the word "offset" appears in this document.  We should probably define what we're talking about here